src/Stack/Dot.hs

```diff
@@ -140,28 +140,37 @@ listDependencies opts = do
   if listDepsTree opts then
     do
       liftIO $ Text.putStrLn "Packages"
-      liftIO $ printTree opts 0 [] pkgs resultGraph
+      liftIO $ printTree opts 0 [] (treeRoots opts pkgs) resultGraph
     else
       void (Map.traverseWithKey go (snd <$> resultGraph))
       where go name payload = liftIO $ Text.putStrLn $ listDepsLine opts name payload
 
+treeRoots :: ListDepsOpts -> Set PackageName -> Set PackageName
+treeRoots opts defaults = let targets = dotTargets $ listDepsDotOpts opts
```

Comment:
These `defaults` are actually the set of local packages, right? Calling them `defaults` obscures the reason for ignoring them in the `else`.

Reply:
That ambiguity makes sense — thank you for pointing it out. I meant whatever `localNames` means here.

https://github.com/commercialhaskell/stack/blob/1bbed48cf22ecc7a3fa3ed230e1cc6354fdf809f/src/Stack/Dot.hs#L89-L104

So this should be changed to `projectNames` too, then?

Reply:
that would be a better name but it's a bit confusing too, I think something like `projectPackages` or `projectPackageNames` would be better, also it makes sense to change the haddock above as well

Reply:
`projectPackages` is clear, as they are clearly names from the types.

I'll keep a look out for any `local`s.